parquet-column/src/main/java/org/apache/parquet/schema/LogicalTypeAnnotation.java

```diff
@@ -1091,6 +1112,104 @@ public int hashCode() {
     }
   }
 
+  public enum GeometryEncoding {
+    WKB
+  }
+
+  public enum Edges {
+    PLANAR,
+    SPHERICAL
+  }
+
+  public static class GeometryLogicalTypeAnnotation extends LogicalTypeAnnotation {
+    private final GeometryEncoding encoding;
+    private final Edges edges;
+    private final String crs;
+    private final ByteBuffer metadata;
+
+    private GeometryLogicalTypeAnnotation(GeometryEncoding encoding, Edges edges, String crs, ByteBuffer metadata) {
+      Preconditions.checkArgument(encoding != null, "Geometry encoding is required");
+      Preconditions.checkArgument(edges != null, "Geometry edges is required");
+      this.encoding = encoding;
+      this.edges = edges;
+      this.crs = crs;
+      this.metadata = metadata;
+    }
+
+    @Override
+    @Deprecated
+    public OriginalType toOriginalType() {
+      return null;
+    }
+
+    @Override
+    public <T> Optional<T> accept(LogicalTypeAnnotationVisitor<T> logicalTypeAnnotationVisitor) {
+      return logicalTypeAnnotationVisitor.visit(this);
+    }
+
+    @Override
+    LogicalTypeToken getType() {
+      return LogicalTypeToken.GEOMETRY;
+    }
+
+    @Override
+    protected String typeParametersAsString() {
+      StringBuilder sb = new StringBuilder();
+      sb.append("(");
+      sb.append(encoding);
+      sb.append(",");
+      sb.append(edges);
+      if (crs != null && !crs.isEmpty()) {
+        sb.append(",");
+        sb.append(crs);
+      }
+      if (metadata != null) {
+        sb.append(",");
+        sb.append(metadata);
+      }
```

Comment:
There's a high probability that `crs` itself contains comma, so this may introduce ambiguity to the generated type parameters.

`metadata` will be appended as something like `java.nio.HeapByteBuffer[pos=_ lim=_ cap=_]` to the type parameter. Does this have any bad implications?

Reply:
This is definitely something that I need to improve :)